Sources/StreamChatUI/ChatMessageList/StreamLayoutAttributes.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+class StreamLayoutAttributes: UICollectionViewLayoutAttributes {
+    var layoutOptions: ChatMessageLayoutOptions?
+    var appearingOptions: ChatMessageLayoutOptions?
+    var disappearingOptions: ChatMessageLayoutOptions?
+    
+    override func copy(with zone: NSZone? = nil) -> Any {
+        let copy = super.copy(with: zone) as! StreamLayoutAttributes
+        copy.layoutOptions = layoutOptions
+        copy.appearingOptions = appearingOptions
+        copy.disappearingOptions = disappearingOptions
+        return copy
+    }
+
+    override func isEqual(_ object: Any?) -> Bool {
+        if let customAttributes = object as? StreamLayoutAttributes {
+            if layoutOptions != customAttributes.layoutOptions
+                || appearingOptions != appearingOptions
+                || disappearingOptions != disappearingOptions
+            { return false }
```

Comment:
When comparing `appearing/disappearing` options both are taken from the lhs 🤔 

I'd also switch to `guard`:
```swift
guard 
 let customAttributes = object as? StreamLayoutAttributes,
 layoutOptions == customAttributes.layoutOptions,
 appearingOptions == customAttributes.appearingOptions,
 disappearingOptions == customAttributes.disappearingOptions
 else { return false }

return super.isEqual(object)

```